src/backend/executor/execMain.c

```diff
@@ -576,6 +576,12 @@ ExecCheckRTPerms(List *rangeTable, bool ereport_on_violation)
 		RangeTblEntry *rte = (RangeTblEntry *) lfirst(l);
 
 		result = ExecCheckRTEPerms(rte);
```

Comment:
I wonder if it makes more sense if add the hook in this function? 
Since it is one returning true and false. 
And for our case, we simplify want to return true in certain cases.

Reply:
Yes, it makes sense to move this check in ExecCheckRTEPerms. Will address this change in next revision.